src/array/cuda/spmm.cu

```diff
@@ -519,7 +519,7 @@ void SpMMCsrHetero(const std::string& op, const std::string& reduce,
              const std::vector<NDArray>& out_aux,
              const std::vector<dgl_type_t>& ufeat_ntids,  // ufeat node type id
              const std::vector<dgl_type_t>& out_ntids) {  // output node type id
-  bool is_scalar_efeat = vec_efeat.size() != 0;
+  bool is_scalar_efeat = vec_efeat[0].NumElements() == vec_csr[0].indices->shape[0];
```

Comment:
Can I use `vec_efeat[0].ndim == 1`?

Reply:
Whats the logic here?

Reply:
Never mind.  `vec_efeat[0].NumElements()` equals `vec_csr[0].indices->shape[0]` means that the number of elements in `vec_efeat[0]` is equal to the number of edges.  This will only happen if the rest of the dimensions are all 1 which doesn't necessarily mean `vec_efeat[0]` is 1-dimensional.